src/Sulu/Component/PHPCR/PathCleanup.php

```diff
@@ -35,14 +35,9 @@ class PathCleanup implements PathCleanupInterface
      */
     private $pattern = '/^(\/[a-z0-9][a-z0-9-_]*)+$/';
 
-    /**
-     * PathCleanup constructor.
-     *
-     * @param array $replacers
-     */
-    public function __construct(array $replacers)
+    public function __construct(SluggerInterface $slugger)
     {
-        $this->replacers = $replacers;
+        $this->slugger = $slugger;
```

Comment:
I think that's where @alexander-schranz mentioned his different approach. If we really know about code directly using this class, we should really handle this case, although I think it is very bad practice...

But I would handle it differently than alex suggested. I would make the parameter in this constructor nullable, and fallback to instantiate the `Slugger` ourself. But I would throw a [deprecation message](https://symfony.com/doc/current/components/phpunit_bridge.html#trigger-deprecation-notices) in that.

Reply:
Sounds good for me:

@Warxcell so this would look like something this to keep the backward compatibility:

```php
public function __construct(array $replacers, SluggerInterface $slugger = null)
{
    if (!$slugger) {
            @trigger_error('Initializing the PathCleanup without a slugger is deprecated since Sulu 2.1.', E_USER_DEPRECATED);
            $slugger = new ASSCISlugger();
    }

    $this->replayers = $replacers;
    $this->slugger = $slluger;
}
```